jina/drivers/craft.py

```diff
@@ -64,7 +64,7 @@ def _apply(self, doc: 'jina_pb2.Document', *args, **kwargs):
             for r in ret:
                 c = doc.chunks.add()
                 self.set_doc_attr(c, r, self._protected_fields)
-                c.length = len(ret)
+                c.length = 0
```

Comment:
why is this zero? `len(ret)` is correct. the idea is to keep track of how many siblings (i.e. other `Document` belongs to the same parent at the same level) this chunk has, which can later be used for normalization in the ranking.

Reply:
It feels weird, to set length to 0, when in documentation it says that refers to the number of chunks. Besides this, this change  not breaking any tests means that we are missing some tests.

Reply:
let's revert it back on this line but instead change the proto docstring to `total number of siblings of this document`. In the next PR, we can think about if there is a better name instead of `length`

```suggestion
                c.length = len(ret)
```

Reply:
ok, let me revert it back together with the test

Reply:
then the assigning of `doc.length = len(ret)` must be wrong?

Reply:
hmm...let me check how `length` is used now

Reply:
I think is not used anywhere

Reply:
`length` is mostly used in `executors/rankers/tfidf.py` (`TfIdfRanker`) as normalizer

Reply:
ok, I see. But one of the 2 usages that is assigned in this PR must be wrong right? They are contradictory in the meaning of `length`. Should it mean number of `children` or number of `siblings`?

Reply:
From the description of `tf-idf` it should be `number of siblings` I think. So we should remove the assignment `doc.length = len(ret)`